core/lib/utils/mail/mail_address.dart

```diff
@@ -137,6 +145,54 @@ class MailAddress with EquatableMixin {
     return localPart;
   }
 
+  String? getLocalPartDetails() {
+    int separatorPosition = localPart.indexOf(subaddressingLocalPartDelimiter);
+    if (separatorPosition <= 0) {
+      return null;
+    }
+    return localPart.substring(separatorPosition + subaddressingLocalPartDelimiter.length);
+  }
+
+  String getLocalPartWithoutDetails() {
+    int separatorPosition = localPart.indexOf(subaddressingLocalPartDelimiter);
+    if (separatorPosition <= 0) {
+      return localPart;
+    }
+    return localPart.substring(0, separatorPosition);
+  }
+
+  MailAddress stripDetails() {
+    return MailAddress(localPart: getLocalPartWithoutDetails(), domain: domain);
+  }
+
+  String strictUrlEncode(String input) {
+    final StringBuffer encoded = StringBuffer();
+
+    for (int codeUnit in utf8.encode(input)) {
+      if ((codeUnit >= 0x30 && codeUnit <= 0x39) || // 0-9
+          (codeUnit >= 0x41 && codeUnit <= 0x5A) || // A-Z
+          (codeUnit >= 0x61 && codeUnit <= 0x7A)) { // a-z
+        encoded.writeCharCode(codeUnit);
+      } else {
+        encoded.write('%${codeUnit.toRadixString(16).padLeft(2, '0').toUpperCase()}');
+      }
+    }
+    return encoded.toString();
+  }
```

Comment:
i tried it with `Uri.encodeComponent()` at first, but the characters `-_.!~*'()` are not encoded, that's why i rewrote it entirely.

Reply:
Where is the upstream repo for this method?
Did we open a bug report?
And maybe even an upstream PR too?
And we should put the link of all this upstream work here.

Reply:
this method i wrote myself

you can find [Uri.encodeComponent() here](https://github.com/dart-lang/sdk/blob/808fa4ca8b5fbc3d79ade02f607b8533c9714029/sdk/lib/core/uri.dart), where they specify that `-_.!~*'()` are left as-is because it's meant to use in the context of RFC 2396.
so there is not really a bug to report right ?

Reply:
>  so there is not really a bug to report right ?

Maybe not then but at least a code comment.